From 1b889dff97d57821226754701ae32c3dc4de6a83 Mon Sep 17 00:00:00 2001
From: Subbaraya Sundeep <sbhatta@marvell.com>
Date: Mon, 16 Sep 2019 19:02:15 +0530
Subject: [PATCH 09/15] octeontx2-pf: Clear mbox interrupt status early

commit a256537dfe79877f69b224b19a25df5509233695 from
git@git.assembla.com:cavium/WindRiver.linux.git

As long as ISR is not cleared upcoming interrupts
will be lost hence clear the ISR first and queue
work for processing.

Change-Id: Ifd68bbbe16b91bf5640d8c7b19478f24c205da6e
Signed-off-by: Subbaraya Sundeep <sbhatta@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/15919
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
index 3cd127cc0f65..ee36e7bd437e 100644
--- a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
+++ b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
@@ -941,12 +941,12 @@ static irqreturn_t otx2_pfaf_mbox_intr_handler(int irq, void *pf_irq)
 	struct otx2_nic *pf = (struct otx2_nic *)pf_irq;
 	struct mbox *mbox;
 
-	mbox = &pf->mbox;
-	otx2_queue_work(mbox, pf->mbox_wq, 0, 1, 1, TYPE_PFAF);
-
 	/* Clear the IRQ */
 	otx2_write64(pf, RVU_PF_INT, BIT_ULL(0));
 
+	mbox = &pf->mbox;
+	otx2_queue_work(mbox, pf->mbox_wq, 0, 1, 1, TYPE_PFAF);
+
 	return IRQ_HANDLED;
 }
 
-- 
2.17.1

